sonoff/xsns_47_max31865.ino

```diff
@@ -0,0 +1,136 @@
+/*
+  xsns_39_MAX31865.ino - MAX31865 thermocouple sensor support for Sonoff-Tasmota
+
+  Copyright (C) 2019 Alberto Lopez Siemens
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_MAX31865
+#include "Adafruit_MAX31865.h"
+
+#define XSNS_47              47
+
+#if MAX31865_PTD_WIRES == 4
+  #define PTD_WIRES MAX31865_4WIRE
+#elif MAX31865_PTD_WIRES == 3
+  #define PTD_WIRES MAX31865_3WIRE
+#else
+  #define PTD_WIRES MAX31865_2WIRE
+#endif
+
+int8_t init_status = 0;
+
+Adafruit_MAX31865 max31865;
+
+struct MAX31865_Result_Struct {
+    uint8_t   ErrorCode;
+    uint16_t  Rtd;
+    float     PtdResistance;
+    float     PtdTemp;
+} MAX31865_Result;
+
+void MAX31865_Init(void){
+    if(init_status)
+        return;
+
+  max31865.setPins(
```

Comment:
Have you thought about using the unchanged lib and instead of a static object use a pointer and allocate the object dynamically here?
(I can provide code if you are unfamiliar with that/don’t know what I mean)

Reply:
Hmm... I personally try to avoid dynamic memory allocation on embedded programming...  

I'm not familiar with Arduino/ESP environment but if experienced people tells me malloc's are safe I have no problem on modifying code. 

On the other hand, the customized library allowed me to add a few more methods: rtd_to_temperature and rtd_to_resistance.